roles/kubespray-defaults/tasks/main.yaml

```diff
@@ -22,3 +22,19 @@
     - no_proxy is not defined
   tags:
     - always
+
+- name: Check if `etcd_kubeadm_enabled` is still in use
+  block:
+    - name: Warn the user if they are still using `etcd_kubeadm_enabled`
+      debug:
+        msg: "WARNING! => `etcd_kubeadm_enabled` is deprecated and will be removed in a future release. You can set `etcd_deployment_type` to `kubeadm`"
+      failed_when: true
+
+    - name: Set `etcd_deployment_type` to "kubeadm" if `etcd_kubeadm_enabled` is true
+      set_fact:
+        etcd_deployment_type: kubeadm
+  when:
+    - etcd_kubeadm_enabled is defined and etcd_kubeadm_enabled
+  ignore_errors: true
```

Comment:
Why here needs to specify ignore_errors: true?
I could not find cases require it.

Reply:
Hello,
Thanks for the review.

I didn't want to fail the playbook for this change. The current implementation warns the user and moves on by setting `etcd_deployment_type`.

I initially put the `ignore_errors` statement below the `failed_when` task, but that didn't make `ansible-lint` happy.

```
roles/kubespray-defaults/tasks/main.yaml:28: ignore-errors Use failed_when and specify error conditions instead of using ignore_errors
``` 

Should I warn the user and fail the playbook run instead?

Reply:
Even if setting etcd_kubeadm_enabled by users and outputting the warning message of line 30, I guess any expected errors don't happen.
I might miss something.

Reply:
The warning message would be better addressed in [roles/kubernetes/preinstall/tasks/0020-verify-settings.yml](https://github.com/kubernetes-sigs/kubespray/pull/8317/files#diff-54d7111eddbe3748d3402dfd1793aacc4981ea9d9cf25f7b614b9dd08c44febc) rather than here and have this block just do the `set_fact` step for backward compatibility.

You can add a #todo comment to clean this when we drop backward compatibility support.

Reply:
Thank you for the reviews. It makes sense to put the check into verify settings. I was torn between choosing a place but haven't thought about putting those tasks into separate files. 😅 

I'm trying to find a better way to alert the user. Throwing a fail stops the playbook, but if I don't fail the task and ignore it, the warning is easy to miss due to being green. I can think of two possible ways and would like to get some guidance on how to resolve this. I couldn't find a good way to issue warnings using Ansible. ☹️ 

This one is just a simple debug task, it shows up as yellow due to `changed_when`
```yaml
- name: Warn the user if they are still using `etcd_kubeadm_enabled`
  debug:
    msg: "WARNING! => `etcd_kubeadm_enabled` is deprecated and will be removed in a future release. You can set `etcd_deployment_type` to `kubeadm` instead of setting `etcd_kubeadm_enabled` to `true`."
  changed_when: etcd_kubeadm_enabled
  run_once: yes
  when: etcd_kubeadm_enabled is defined
```

We can use the following assert task in the future when we decide to drop the compatibility support.
```yaml
- name: Warn the user if they are still using `etcd_kubeadm_enabled`
  assert:
    that: not etcd_kubeadm_enabled
    msg: "WARNING! => `etcd_kubeadm_enabled` is deprecated and will be removed in a future release. You can set `etcd_deployment_type` to `kubeadm` instead of setting `etcd_kubeadm_enabled` to `true`."
  changed_when: true
  failed_when: false
  run_once: yes
  when: etcd_kubeadm_enabled is defined
```

Or should we drop the compatibility support right now and force users to change the variable? It could mess up with someone's automation but should not be a huge issue overall.

cc: @cristicalin @oomichi (Sorry for the second ping. I somehow submitted the previous comment as a review. GitHub deleted the comment itself when I tried to delete the review. 😅)

Reply:
Personally I would not drop backward compatibility at this stage (see the mess I caused with docker to containerd as an example of the fallout).

You can assert any incompatible setting which should stop the playbook like `etcd_kubeadm_enabled` set to `True` and `etcd_deployment_type` != `kubeadm`. And then output a warning with `debug` if `etcd_kubeadm_enabled` is still, unfortunately this is an ansible limitation and I'm not aware of a better way to output deprecation messages in ansible than just using `debug`. Happy to learn though.

Reply:
I have added an assertion to check incompatible settings. It will lay dormant for a while though since we run `kubespray-defaults` before assertions. However, it might be useful in the future, when we decide to deprecate `etcd_kubeadm_enabled`.

I also added `changed_when: true` to debug task, so it will be a bit more noticeable.